Alignment/CTPPS/src/StraightTrackAlignment.cc

```diff
@@ -327,20 +327,61 @@ void StraightTrackAlignment::ProcessEvent(const DetSetVector<TotemRPRecHit> &hit
   
   HitCollection hitSelection;
 
-  // strips
-  for (const auto &pv : hitsStrip)
+// strips (With UV pattern as input)
+
+  for (const auto &pv : UVPattern)
   {
-    // skip if RP not selected
     const CTPPSDetId detId(pv.detId());
     const unsigned int rpDecId = detId.arm()*100 + detId.station()*10 + detId.rp();
 
     if (find(rpIds.begin(), rpIds.end(), rpDecId) == rpIds.end())
       continue;
 
-    const double &z = task.geometry[pv.detId()].z;
+    // is U-V association unique?
+    unsigned int n_U=0, n_V=0;
+    unsigned int idx_U=0, idx_V=0;
+    for (unsigned int pi = 0; pi < pv.size(); pi++)
+    {
+      const TotemRPUVPattern &pattern = pv[pi];
 
-    for (const auto &h : pv)
-      hitSelection.emplace_back(Hit(pv.detId(), 2, h.getPosition(), h.getSigma(), z));
+       switch (pattern.getProjection())
+      {
+        case TotemRPUVPattern::projU:
+          n_U++;
+          idx_U=pi;
+          break;
+     
+        case TotemRPUVPattern::projV:
+          n_V++;
+          idx_V=pi;
+          break;
+
+        default:
+          break;
+       }  
+    }
+    //TODO Uncomment for real data.
```

Comment:
Please remove this block. We will always use only events with 1U and 1V pattern.